python/ray/tune/trainable.py

```diff
@@ -306,32 +309,42 @@ def _train(self):
     def _save(self, checkpoint_dir):
         """Subclasses should override this to implement save().
 
+        See also: ray.tune.Trainable.save_dict.
+
         Args:
             checkpoint_dir (str): The directory where the checkpoint
                 can be stored.
 
         Returns:
-            Checkpoint path that may be passed to restore(). Typically
-                would default to `checkpoint_dir`.
+            checkpoint_path: The checkpoint path that will be
+                passed to restore(). This can be different from
+                checkpoint_dir.
```

Comment:
Do we actually have a use case for it being different? It would be nice to add a check that checkpoint_path is a child of checkpoint_dir.